src/stackdriver-nozzle/nozzle/metric_sink.go

```diff
@@ -47,38 +49,41 @@ func (ms *metricSink) Receive(envelope *events.Envelope) error {
 		int64(timestamp%time.Second),
 	)
 
-	var metrics []stackdriver.Metric
+	var metrics []messages.Metric
 	switch envelope.GetEventType() {
 	case events.Envelope_ValueMetric:
 		valueMetric := envelope.GetValueMetric()
-		metrics = []stackdriver.Metric{{
+		metrics = []messages.Metric{{
 			Name:      valueMetric.GetName(),
+			Type:      envelope.GetEventType(),
```

Comment:
I probably missed this somewhere else in the CL, but what does the new `Type` field get us?

Reply:
I needed it in in the [router](https://github.com/cloudfoundry-community/stackdriver-tools/pull/117/commits/b289b73e42b1284685f838e509f7db6dec42fa7d#diff-b5bfcc892bab6eeaa9fcaf2add9391bbR36) to determine where to send it. I'm not crazy about it because it adds a specific type from loggregator into the internal representation. If you have other ideas I'm all ears.